### ISSUE TITLE
Multi-line body

### ISSUE THREAD

**Issue by Seluj78** [JasonEtco/todo#93]
I've noticed it is not possible to do a multiline body with **todo-bot**

Would it be possible to add it as a feature ? for example, here's a comment i placed in my code : 

```python
    """ 
        @todo Add option priority explaination in the wiki 
        @body If page_number is explicitely set through options when calling test_get, then the return will be the page PAGE_NUMBER. If from_page and to_page are set, it will loop through them Else, it will do all the pages until return is empty. priority on page_number. 1800 requetes max pour to_page car 1800 requete en limite de l'api. a tester 
    """ 
```
This really long body is horrible to read and I would've loved to have it in a multi line body :)
(BTW, this is one of the todos that didn't pass in my latest commit (CC #89 ))

**Comment by Seluj78** [comment]
Okay, after deleting and putting the todo bacl, your bot made the issue and here it is, without the multiline comment : https://github.com/Seluj78/IntraPy/issues/28

**Comment by Seluj78** [comment]
Here, I formatted it manually : https://github.com/Seluj78/IntraPy/issues/28#issuecomment-358276285

But it would be something i'd like to be able to do within the code

**Comment by JasonEtco** [comment]
I would _love_ to do this. It's not very simple unfortunately, but I agree that it would be great.

Although FWIW, writing really large bodies is usually against what **todo** is trying to help with, which is quickly (emphasis on quickly) creating notes for yourself without leaving the context of your code.

**Comment by Seluj78** [comment]
Alright ! Works for me, I'll see what I can do, and how I can solve this problem on my side

**Comment by rsarky** [comment]
Hey . I was going through this bug and had a few questions.
Should this action be the default for all cases?
For eg. If we have a body with 2 sentences,should the bot write a 2 line comment?
Or should there be a cut off number of sentences after which the bot multi lines the comment

**Comment by JasonEtco** [comment]
@rsarky That's an interesting question. In YAML, multi-line comments must be prefixed with a `>` character and go until the next property; what do you think about going to the next empty new line? So:

```js
// BODY > This
// is
// a
// comment

// and this isn't part of it because of the empty line ^
```

This means that there won't be a max number of lines (although there probably should be, just in case), and it leaves it a little more open. The drawback is that some folks won't want an empty line right after their comment. What do you think?

**Comment by rsarky** [comment]
I like the idea.
So what you propose is :
```
@todo : This is a sample comment.
@body : > Sample comment line 1.
Sample comment line 2.
Sample comment line 3.

Multline ends here
```
We could consider a special character or group of characters instead of an empty line to end the multiline if empty line is not desirable?

**Comment by JasonEtco** [comment]
> We could consider a special character or group of characters instead of an empty line to end the multiline if empty line is not desirable?

Totally! That'd have to be something pretty obvious, but still small enough to not be annoying to look at in the code. It'd also need to ignore the entire line that that end character(s) is on, otherwise it'd also consider things like comment delimiters as part of the body.

There is another significant problem though; different languages have different comment syntax, and we don't want that to come into the body. For example:

```js
// BODY > Here
// and here
```

The body should be:

```
Here
and here
```

But unless there's some real smart comment syntax detection, it'd render as:

```
Here
// and here
```

**Comment by rsarky** [comment]
```
Here
// and here
```
Fixing this would require us to write code specific to different languages and I dont think that is feasible.
Consider this. The user writes the comment in a single line. And whenever he/she wants a new line they insert a token like ```<br>``` or somethine along the same lines.
Eg.
```
// BODY : Here <br> Here.
```
The body should be:
```
Here
Here.
```

**Comment by JasonEtco** [comment]
@rsarky using `<br>` tags should actually work already. Worth trying out to be certain, but that shouldn't require any more work; if thats the case, it should be documented that thats a possibility 📖

**Comment by rsarky** [comment]
I will test it to be sure.If it works I will document it.

**Comment by rsarky** [comment]
I did a bit of testing the ```<br>``` tags aren't working. But I guess they aren't always supposed to work in markdown?
AFAIK the markdown way of adding a new line is adding two spaces after you end a new line then you go to a new line.
I tested doing the above too but it doesn't seem to work. It just appends everything into one line

**Comment by JasonEtco** [comment]
@rsarky - **todo** doesn't do any markdown parsing; it's all left to GitHub. In other words, `<br>` works fine to start a new line on GitHub (you can try it out in an issue comment), but it isn't being sent correctly by **todo**, probably because `Handlebars` auto-escapes "dangerous" characters. I'm cool with removing the safety check there, provided we add some more validation so that its still "safe."

**Comment by rsarky** [comment]
I think `https://stackoverflow.com/q/12331077/7570841` is exactly what we want?

**Comment by JasonEtco** [comment]
Note quite; Handlebars will turn `<br>` into `&lt;br&gt;`, so we'd just need to either turn it back:

```js
text.replace('&lt;br&gt;', '<br>')
```

Or don't sanitize the input at all (notice the triple curly braces instead of the normal two):

```md
{{{ body }}}
```

The former is an extra calculation that the app has to do (not a huge deal), and the latter presents an a way for folks to inject malicious code. So I'm a 👍 to the first option!

**Comment by rsarky** [comment]
LGTM.
I will start working on it and submit a PR asap. :+1:

**Comment by JasonEtco** [comment]
Awesome @rsarky! You should look in `/src/utils/parse-diff.js` 😄

**Comment by rsarky** [comment]
>npm start
\> todo@1.0.0 start /home/rohit/internship/gsoc/todo
\> PRIVATE_KEY=$(echo $PRIVATE_KEY | base64 -d) npm run start-probot
\> todo@1.0.0 start-probot /home/rohit/internship/gsoc/todo
\> probot run ./src/index.js
module.js:540
    throw err;
Error: Cannot find module '../src/private-key'
    at Function.Module._resolveFilename (module.js:538:15)
    at Function.Module._load (module.js:468:25)
    at Module.require (module.js:587:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/home/rohit/internship/gsoc/todo/node_modules/probot-ts/bin/probot-run.js:8:26)
    at Module._compile (module.js:643:30)
    at Object.Module._extensions..js (module.js:654:10)
    at Module.load (module.js:556:32)
    at tryModuleLoad (module.js:499:12)
    at Function.Module._load (module.js:491:3)

This may be a silly question but I am getting the following error when I run npm start.
So I need to make a private_key file inside the src folder and export `PRIVATE_KEY` ?
I guess this is required for jest to function properly?

**Comment by JasonEtco** [comment]
@rsarky ah sorry that's my fault for overcomplicating it; you can run `npm run probot-start` to try it out locally. That's an artifact from an older deployment setup I had, should be removed.

**Comment by rsarky** [comment]
:face_with_head_bandage:  Should have asked earlier instead of racking my brain over it :laughing:

**Comment by rsarky** [comment]
Although `npm run probot-start` is still throwing the same error.

**Comment by JasonEtco** [comment]
@rsarky feel free to jump into the [Probot Slack](https://probot-slackin.herokuapp.com/) if you want some quicker feedback/question-answering 👍

**Comment by JasonEtco** [comment]
You do need a `.pem` Private Key file; follow the [setup instructions](https://probot.github.io/docs/development/#configure-a-github-app) to register a GitHub App and get all that authentication stuff sorted 👌

**Comment by JasonEtco** [comment]
Closed by #112 🎉
